src/test/ec2/prompter.test.ts

```diff
@@ -0,0 +1,181 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import * as assert from 'assert'
+import { Ec2Prompter, instanceFilter } from '../../ec2/prompter'
+import { Ec2Instance } from '../../shared/clients/ec2Client'
+import { RegionSubmenuResponse } from '../../shared/ui/common/regionSubmenu'
+import { Ec2Selection } from '../../ec2/utils'
+import { AsyncCollection } from '../../shared/utilities/asyncCollection'
+import { intoCollection } from '../../shared/utilities/collectionUtils'
+import { DataQuickPickItem } from '../../shared/ui/pickerPrompter'
+
+describe('Ec2Prompter', async function () {
+    class MockEc2Prompter extends Ec2Prompter {
+        public instances: Ec2Instance[] = []
+
+        public testAsQuickPickItem(testInstance: Ec2Instance) {
+            return Ec2Prompter.asQuickPickItem(testInstance)
+        }
+
+        public testGetSelectionFromResponse(response: RegionSubmenuResponse<string>): Ec2Selection {
+            return Ec2Prompter.getSelectionFromResponse(response)
+        }
+        public async testGetInstancesAsQuickPickItems(region: string): Promise<DataQuickPickItem<string>[]> {
```

Comment:
Can get rich testing of wizard behavior using `wizardTestUtils`. 

See `src/test/cloudWatchLogs/commands/searchLogGroup.test.ts`, here's a commit that migrated the tests to use `wizardTestUtils`:

https://github.com/aws/aws-toolkit-vscode/commit/8eed0da9990b0b699faa50f9125a71d8854ddca8

Reply:
The relevant wizard testing lives in `src/test/shared/ui/prompters/regionSubmenu.test.ts`. This test file is aimed at testing the utility functions surrounding the prompter, since the prompter is just implementing the already tested component of `regionSubmenu`.